runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/graph/ImmutableExecutableStage.java

```diff
@@ -21,27 +21,53 @@
 import com.google.auto.value.AutoValue;
 import com.google.common.collect.ImmutableSet;
 import java.util.Collection;
+import java.util.stream.Collectors;
+import org.apache.beam.model.pipeline.v1.RunnerApi;
 import org.apache.beam.model.pipeline.v1.RunnerApi.Environment;
 import org.apache.beam.runners.core.construction.graph.PipelineNode.PCollectionNode;
 import org.apache.beam.runners.core.construction.graph.PipelineNode.PTransformNode;
 
 /** An {@link ExecutableStage} which is constructed with all of its initial state. */
 @AutoValue
 abstract class ImmutableExecutableStage implements ExecutableStage {
+  static ImmutableExecutableStage ofFullComponents(
+      RunnerApi.Components components,
```

Comment:
We're importing nested classes directly elsewhere in this class.

Reply:
Done.